pkg/server/cluster_server.go

```diff
@@ -42,23 +52,61 @@ type clusterServer struct {
 // It accepts the kubeConfig which is not required when it's
 // run from within the cluster(useful in testing).
 // It accepts the apiserverURL which is the location of the KubeAPIServer.
-func NewClusterServer(kubeConfig, apiserverURL string) (Server, error) {
+func NewClusterServer(kubeConfig, apiserverURL string, enableMaster bool) (Server, error) {
 	restConfig, err := getClientConfig(kubeConfig)
 	if err != nil {
 		return nil, fmt.Errorf("Failed to create Kubernetes rest client: %v", err)
 	}
 
+	kc, err := clientset.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
 	mc := v1.NewForConfigOrDie(restConfig)
 	return &clusterServer{
+		kubeClient:     kc,
 		machineClient:  mc,
 		kubeconfigFunc: func() ([]byte, []byte, error) { return kubeconfigFromSecret(bootstrapTokenDir, apiserverURL) },
+		enableMaster:   enableMaster,
 	}, nil
 }
 
 // GetConfig fetches the machine config(type - Ignition) from the cluster,
 // based on the pool request.
-func (cs *clusterServer) GetConfig(cr poolRequest) (*ignv2_2types.Config, error) {
-	mp, err := cs.machineClient.MachineConfigPools().Get(cr.machineConfigPool, metav1.GetOptions{})
+func (cs *clusterServer) GetConfig(cr poolRequest, auth string) (*ignv2_2types.Config, error) {	
+	authSecretObj, err := cs.kubeClient.CoreV1().Secrets(componentNamespace).Get(ignitionAuth, metav1.GetOptions{})
```

Comment:
unit test failing here, I think:

```panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xf8 pc=0xec8d91]

goroutine 85 [running]:
testing.tRunner.func1(0xc4203d9ef0)
	/usr/local/go/src/testing/testing.go:742 +0x29d
panic(0xfcf0e0, 0x18ee930)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/openshift/machine-config-operator/pkg/server.(*clusterServer).GetConfig(0xc420499f38, 0x1101419, 0x9, 0x0, 0x0, 0x0, 0x0, 0xf)
	/go/src/github.com/openshift/machine-config-operator/pkg/server/cluster_server.go:77 +0x41
github.com/openshift/machine-config-operator/pkg/server.TestClusterServer(0xc4203d9ef0)
	/go/src/github.com/openshift/machine-config-operator/pkg/server/server_test.go:175 +0x936
testing.tRunner(0xc4203d9ef0, 0x117d5a0)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0
FAIL	github.com/openshift/machine-config-operator/pkg/server	0.039s```

Reply:
Yeah, had forgotten to `git commit --amend` before pushing.  Fixed already in the latest :arrow_down: